docs/api/specs/browserResidentDesktopAgents.md

```diff
@@ -0,0 +1,253 @@
+  - Note that the `pagehide` event may not fire if the window's render thread crashes or is closed while 'frozen'.
+- By polling the application for responses via the `heartbeatEvent` and `heartbeatAcknowledgement` messages provided in the [Desktop Agent Communication Protocol](./desktopAgentCommunicationProtocol#checking-apps-are-alive). These message may be used for both periodic and on-demand polling by DA implementations. On-demand polling could, for example, be used to check that all instances returned in a findIntent response or displayed in an intent resolver are still alive.
+  - Desktop Agents MAY determine their own timeout, or support configuration, to be used for considering an application to have closed as this may be affected by the implementation details of app and DAs.
+
+Finally, Desktop Agents SHOULD retain instance details for applications that have closed as they may appear to close during navigation events, or may navigate away and then navigate back. By retaining the instance data (`instanceId`, `instanceUuid` and `WindowProxy`) the same instance identity can be maintained or reissued. There is no standard length of time that such details should be retained, hance, Desktop Agents MAY determine for themselves how long to retain instance details for closed instances.
+
+## Responding to app communications with Desktop Agent Communication Protocol (DACP)
+
+After validating an application's identity and any instance identity to be reused, the Desktop Agent is ready to support communication with the application. This is achieved via the [Desktop Agent Communication Protocol](./desktopAgentCommunicationProtocol) (DACP) over the `MessageChannel` established in the previous steps.
+
+## Implementing DAs in hidden iframes
+
+As described above, DA providers can leverage hidden iframes to establish a communication mechanism that is independent of a Parent window. This approach allows apps to connect to a DA even when they were not opened by that DA.
+
+The hidden iframe url can be provided in two ways:
+
+1. By a Parent window - This allows DAs to redirect communications to via a hidden iframe that loads a known URL. The main benefit of this approach is that it can allow a system to continue to operate even if the parent window is closed.
+2. By a `failover` function - When no parent DA can be found (such as when a tab is opened directly by an end user) then a failover function can create a hidden iframe and return a reference to it (a `WindowProxy`) to initiate communication with via the WCP and DACP in the same way as we do with a parent window. Alternatively, a `DesktopAgent` implementation maybe loaded directly and returned from the failover function, which `getAgent()` will pass-through.
```

Comment:
`to initiate communication with via the WCP` seems wrong to me (feel free to ignore if this is intended. I think this should either be to `initiate communication via the WCP` or to `initiate communication with the WCP`

Reply:
also -
```
 a `DesktopAgent` implementation maybe loaded
```
=> 
```
 a `DesktopAgent` implementation may be loaded
```

again - feel free to ignore. English not good mine is.

Reply:
redrafted as:
>...and return a reference to it (a `WindowProxy`) that is used to initiate communication via the WCP in the same way as we do with a parent window or frame.